src/main/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo.java

```diff
@@ -292,7 +292,7 @@
      * @since 2.2
      */
     @Parameter( property = "encoding", defaultValue = "${project.build.sourceEncoding}" )
-    private String encoding;
+    private String inputEncoding;
```

Comment:
breaking change

Reply:
That is correct, have a look at the commit message. With M-R-Impl 3.2.0 this is a problem. I do admit. I haven't really thought about how to solve it since this is a transient issue. What would be your proposal?